drink/src/chain_api.rs

```diff
@@ -1,167 +1,117 @@
 //! Basic chain API.
 
 use frame_support::{
-    sp_runtime::{traits::Dispatchable, DispatchResultWithInfo, Saturating},
-    traits::tokens::currency::Currency,
+    sp_runtime::{traits::Dispatchable, DispatchError, DispatchResultWithInfo, Saturating},
+    traits::fungible::Mutate,
 };
 use frame_system::pallet_prelude::BlockNumberFor;
 
 use crate::{runtime::AccountIdFor, DrinkResult, Error, EventRecordOf, Runtime, Sandbox};
 
+/// Generic Time type.
+pub type MomentOf<R> = <R as pallet_timestamp::Config>::Moment;
+
+/// Generic fungible balance type.
+pub type BalanceOf<R> = <R as pallet_balances::Config>::Balance;
+
 /// The runtime call type for a particular runtime.
 pub type RuntimeCall<R> = <R as frame_system::Config>::RuntimeCall;
 
-/// Interface for basic chain operations.
-pub trait ChainApi<R: Runtime> {
-    /// Return the current height of the chain.
-    fn current_height(&mut self) -> BlockNumberFor<R>;
-
+impl<R: Runtime> Sandbox<R> {
```

Comment:
this block could go to the module with `Sandbox` itself

Reply:
or actually, we could have a module for sandbox instead of this `chain_api` module

Reply:
ah right, will move that where sandbox is defined